src/modelbuilder.pyx

```diff
@@ -0,0 +1,180 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#******************************************************************************/
+
+# We expose DAAL's directly through Cython
+# Currently only GBT is available.
+# The model builder object is retrieved through calling model_builder.
+# We will extend this once we know how other model builders willl work in DAAL
+
+cdef extern from "modelbuilder.h":
+    ctypedef size_t c_gbt_clf_NodeId
+    ctypedef size_t c_gbt_clf_TreeId
+    ctypedef size_t c_gbt_reg_NodeId
+    ctypedef size_t c_gbt_reg_TreeId
+
+    cdef size_t c_gbt_clf_noParent
+    cdef size_t c_gbt_reg_noParent
+
+    cdef cppclass c_gbt_classification_ModelBuilder:
+        c_gbt_classification_ModelBuilder(size_t nFeatures, size_t nIterations, size_t nClasses) except +
+        c_gbt_clf_TreeId createTree(size_t nNodes, size_t classLabel)
+        c_gbt_clf_NodeId addLeafNode(c_gbt_clf_TreeId treeId, c_gbt_clf_NodeId parentId, size_t position, double response)
+        c_gbt_clf_NodeId addSplitNode(c_gbt_clf_TreeId treeId, c_gbt_clf_NodeId parentId, size_t position, size_t featureIndex, double featureValue)
+
+    cdef cppclass c_gbt_regression_ModelBuilder:
+        c_gbt_regression_ModelBuilder(size_t nFeatures, size_t nIterations) except +
+        c_gbt_reg_TreeId createTree(size_t nNodes)
+        c_gbt_reg_NodeId addLeafNode(c_gbt_reg_TreeId treeId, c_gbt_reg_NodeId parentId, size_t position, double response)
+        c_gbt_reg_NodeId addSplitNode(c_gbt_reg_TreeId treeId, c_gbt_reg_NodeId parentId, size_t position, size_t featureIndex, double featureValue)
+
+    cdef gbt_classification_ModelPtr * get_gbt_classification_modelbuilder_Model(c_gbt_classification_ModelBuilder *)
+    cdef gbt_regression_ModelPtr * get_gbt_regression_modelbuilder_Model(c_gbt_regression_ModelBuilder *)
+
+
+cdef class gbt_classification_modelbuilder:
+    '''
+    Model Builder for gradient boosted trees.
+    '''
+    cdef c_gbt_classification_ModelBuilder * c_ptr
+
+    def __cinit__(self, size_t nFeatures, size_t nIterations, size_t nClasses):
+        self.c_ptr = new c_gbt_classification_ModelBuilder(nFeatures, nIterations, nClasses)
+
+    def __dealloc__(self):
+        del self.c_ptr
+
+    def create_tree(self, size_t nNodes, size_t classLabel):
+        '''
+        Create certain tree in the gradient boosted trees classification model for certain class
+
+        :param size_t nNodes: number of nodes in created tree
+        :param size_t classLabel: label of class for which tree is created. classLabel bellows interval from 0 to (nClasses - 1)
+        :rtype: tree identifier
+        '''
+        return self.c_ptr.createTree(nNodes, classLabel)
+
+    def add_leaf(self, c_gbt_clf_TreeId treeId, double response, c_gbt_clf_NodeId parentId=c_gbt_clf_noParent, size_t position=0):
```

Comment:
For variable names, you use `CamelCase`. For method names, you use `snake_case`. Why?

Reply:
I thought that this is a right way, because the model builder for DF, which I used as an example was written like that. it'd be better to use `snake_case` everywhere I guess?

Reply:
I see file https://github.com/IntelPython/daal4py/blob/master/src/gettree.pyx for example.  For me, I need to make out in the same style. It is difficult to review without an example - because it is not clear how it will be used and how it will look. Can we merge after added examples and tests?

Reply:
Examples will be ready soon. The point is that the examples also need examples for themselves which are not ready yet. Of course we can merge that in any order, but there's no much time remains before the code freeze. I'll try to push the first version of another PR till tomorrow's daily, so you could have a look on it. Will that suit you?

Reply:
I've changed all name styles to `snake_case`. It all looks same now